libraries/AC_AutoTune/AC_AutoTune.cpp

```diff
@@ -1728,56 +1282,824 @@ void AC_AutoTune::get_poshold_attitude(float &roll_cd_out, float &pitch_cd_out,
+        tgt_rate_reading = attitude_control->rate_bf_targets().z;
+        if (settle_time > 0) {
+            settle_time--;
+            trim_command_reading = motors->get_yaw();
+            trim_heading = ahrs_view->yaw_sensor;
+        } else if (((wrap_180_cd(ahrs_view->yaw_sensor - trim_heading) <= 2.0f * max_angle_cd && is_positive(dir_sign))
+                   || (wrap_180_cd(ahrs_view->yaw_sensor - trim_heading) >= -2.0f * max_angle_cd && !is_positive(dir_sign)))
+                   && ff_test_phase == 0) {
+            rate_request_cds += (target_rate_cds - rate_request_cds) * filt_alpha;
+            attitude_control->input_rate_bf_roll_pitch_yaw(0.0f, 0.0f, rate_request_cds);
+        } else if (((wrap_180_cd(ahrs_view->yaw_sensor - trim_heading) > 2.0f * max_angle_cd && is_positive(dir_sign))
+                   || (wrap_180_cd(ahrs_view->yaw_sensor - trim_heading) < -2.0f * max_angle_cd && !is_positive(dir_sign)))
+                   && ff_test_phase == 0) {
+            ff_test_phase = 1;
+            rate_request_cds += (-target_rate_cds - rate_request_cds) * filt_alpha;
+            attitude_control->input_rate_bf_roll_pitch_yaw(0.0f, 0.0f, rate_request_cds);
+            attitude_control->rate_bf_yaw_target(rate_request_cds);
+        } else if (((wrap_180_cd(ahrs_view->yaw_sensor - trim_heading) >= -2.0f * max_angle_cd && is_positive(dir_sign))
+                   || (wrap_180_cd(ahrs_view->yaw_sensor - trim_heading) <= 2.0f * max_angle_cd && !is_positive(dir_sign)))
+                   && ff_test_phase == 1 ) {
+            rate_request_cds += (-target_rate_cds - rate_request_cds) * filt_alpha;
+            attitude_control->input_rate_bf_roll_pitch_yaw(0.0f, 0.0f, rate_request_cds);
+            attitude_control->rate_bf_yaw_target(rate_request_cds);
+        } else if (((wrap_180_cd(ahrs_view->yaw_sensor - trim_heading) < -2.0f * max_angle_cd && is_positive(dir_sign))
+                   || (wrap_180_cd(ahrs_view->yaw_sensor - trim_heading) > 2.0f * max_angle_cd && !is_positive(dir_sign)))
+                   && ff_test_phase == 1 ) {
+            ff_test_phase = 2;
+            angle_request_cd = attitude_control->get_att_target_euler_cd().z;
+            attitude_control->input_euler_angle_roll_pitch_yaw(start_angles.x, start_angles.y, angle_request_cd, false);
+        } else if (ff_test_phase == 2 ) {
+            angle_request_cd += wrap_180_cd(trim_heading - angle_request_cd) * filt_alpha;
+            attitude_control->input_euler_angle_roll_pitch_yaw(start_angles.x, start_angles.y, angle_request_cd, false);
+        }
+        break;
+    }
+
+    rotation_rate = rotation_rate_filt.apply(gyro_reading,
+                    AP::scheduler().get_loop_period_s());
+    command_out = command_filt.apply((command_reading - trim_command_reading),
+                                     AP::scheduler().get_loop_period_s());
+    filt_target_rate = target_rate_filt.apply(tgt_rate_reading,
```

Comment:
This line is producing a compile error for me at least.
```
[752/818] Compiling ArduCopter/Copter.cpp
../../libraries/AC_AutoTune/AC_AutoTune.cpp: In member function 'void AC_AutoTune::rate_ff_test_run(float, float, float)':
../../libraries/AC_AutoTune/AC_AutoTune.cpp:1567:46: error: 'tgt_rate_reading' may be used uninitialized in this function [-Werror=maybe-uninitialized]
     filt_target_rate = target_rate_filt.apply(tgt_rate_reading,
                        ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
                        AP::scheduler().get_loop_period_s());
                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.
```

Reply:
@rmackay9  I was recently working on removing some static variables and hadn’t completed the work.  Remove the last commit and it should compile and run fine.